### PR TITLE
Fix compatibility issue for android 16 device.Fixes #22294

### LINKED ISSUE #22294: [BUG] Compatibility issue for android 16
Introduced in https://github.com/facebook/react-native/commit/e02a154787274be1da3632cb1412554cbd53928b.
stacktrace
```bash
11-15 22:34:01.684 1965-1982/? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
    java.lang.RuntimeException: An error occured while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:299)
        at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:273)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:124)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:307)
        at java.util.concurrent.FutureTask.run(FutureTask.java:137)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
        at java.lang.Thread.run(Thread.java:856)
     Caused by: java.lang.NoClassDefFoundError: java.nio.charset.StandardCharsets
        at com.facebook.react.modules.systeminfo.AndroidInfoHelpers.getMetroHostPropValue(AndroidInfoHelpers.java:87)
        at com.facebook.react.modules.systeminfo.AndroidInfoHelpers.getServerIpAddress(AndroidInfoHelpers.java:61)
        at com.facebook.react.modules.systeminfo.AndroidInfoHelpers.getServerHost(AndroidInfoHelpers.java:39)
        at com.facebook.react.packagerconnection.PackagerConnectionSettings.getDebugServerHost(PackagerConnectionSettings.java:42)
        at com.facebook.react.packagerconnection.JSPackagerClient.<init>(JSPackagerClient.java:77)
        at com.facebook.react.devsupport.DevServerHelper$1.doInBackground(DevServerHelper.java:194)
        at com.facebook.react.devsupport.DevServerHelper$1.doInBackground(DevServerHelper.java:151)
        at android.os.AsyncTask$2.call(AsyncTask.java:287)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
        at java.util.concurrent.FutureTask.run(FutureTask.java:137) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569) 
        at java.lang.Thread.run(Thread.java:856) 
```

### PR DESCRIPTION
**Pull request by gengjiawen**
## Motivation
Fix compatibility issue for android 16 device. Related issue https://github.com/facebook/react-native/pull/22295.
## Test Plan
pass all current ci.
## Related PRs
none
## Release Notes
 [GENERAL] [ANDROID] [Fixed] - Fix compatibility issue for android 16 device

### CODE REVIEW

**Comment by gengjiawen** [comment]
@stepanhruda please review.

**Comment by hramos** [comment]
The `test_android` failure should be unrelated. Can you re-run the `test_android` workflow to make sure? There's a button at the top right of the window at 
https://circleci.com/gh/gengjiawen/react-native/1237?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link that can do that.

**Comment by gengjiawen** [comment]
I will run it again.

File: ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
```
Comment (hey99xx) [inline ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/AndroidInfoHelpers.java]: not deleting this line? it won't crash but still incorrect to import it, may be accidentaly used again in the future

**Pull request closed without merging**